### PR TITLE
Bug 1611158 - Subtest view - include the platform in the subtest name

### PR DESCRIPTION
**Pull request by esanuandra**

### CODE REVIEW

**Review by ionutgoldan — COMMENTED** [review]
This is looking good!

What's left here is the test coverage. Could you look for a way to provide it, so this functionality won't break in the future?

File: tests/ui/perfherder/compare-view/compare_page_title_test.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+test('Compare page title updates with subtests name if there are subtests', async () => {
+  const { getByTestId, rerender } = render(comparePageTitle(false));
+
+  let pageTitle = await waitFor(() => getByTestId('compare-page-title'));
+
+  expect(pageTitle.textContent).toBe('Perfherder Compare Revisions');
+
+  rerender(comparePageTitle(true));
+
+  pageTitle = await waitFor(() => getByTestId('compare-page-title'));
```
Comment (ionutgoldan) [inline tests/ui/perfherder/compare-view/compare_page_title_test.jsx:36]: Don't grab the title by `id`, as that's coding against implementation. Query by the title's text directly.
That way, if we'll changed the underlying HTML structure, the tests won't break.

File: tests/ui/perfherder/compare-view/compare_page_title_test.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+afterEach(cleanup);
+
+test('Compare page title updates with subtests name if there are subtests', async () => {
+  const { getByTestId, rerender } = render(comparePageTitle(false));
+
+  let pageTitle = await waitFor(() => getByTestId('compare-page-title'));
+
+  expect(pageTitle.textContent).toBe('Perfherder Compare Revisions');
```
Comment (ionutgoldan) [inline tests/ui/perfherder/compare-view/compare_page_title_test.jsx:32]: I don't see the point of testing this main title. This logic isn't specific to `ComparePageTitle`, but to the client code.

File: tests/ui/perfherder/compare-view/compare_page_title_test.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const comparePageTitle = (hasSubtests) => {
+  return (
+    <ComparePageTitle
+      title={
+        hasSubtests
+          ? `${title} subtest summary`
+          : 'Perfherder Compare Revisions'
+      }
+      pageTitleQueryParam={null}
+      defaultPageTitle={defaultPageTitle}
```
Comment (ionutgoldan) [inline tests/ui/perfherder/compare-view/compare_page_title_test.jsx:20]: We need another test for the logic used for `defaultPageTitle`.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: tests/ui/perfherder/compare-view/compare_page_title_test.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    />
+  );
+};
+
+afterEach(cleanup);
+
+test('Compare page title includes the platform name with subtest summary when there are subtests', async () => {
+  const { getByText } = render(comparePageTitle(true));
+
+  const pageTitle = await waitFor(() => getByText(`${title}subtest summary`));
```
Comment (ionutgoldan) [inline tests/ui/perfherder/compare-view/compare_page_title_test.jsx:30]: Typo, as we forgot space?

File: tests/ui/perfherder/compare-view/compare_page_title_test.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  expect(pageTitle.textContent).toBe(`${title} subtest summary`);
```
Comment (ionutgoldan) [inline tests/ui/perfherder/compare-view/compare_page_title_test.jsx:32]: I think we could just check the existence of an element with this title.

**Review by ionutgoldan — APPROVED** [review]

File: ui/shared/ComparePageTitle.jsx [hunk 6]
```diff
@@ -18,6 +18,20 @@ export default class ComparePageTitle extends React.Component {
     };
   }
 
+  componentDidUpdate(prevProps) {
+    const { pageTitleQueryParam, title } = this.props;
+    if (
+      prevProps.pageTitleQueryParam !== pageTitleQueryParam ||
+      prevProps.title !== title
+    ) {
+      // eslint-disable-next-line react/no-did-update-set-state
```
Comment (sarah-clements) [inline ui/shared/ComparePageTitle.jsx:27]: Rather than disabling this rule, can you find a workaround so you don't need to?

**Review by sarah-clements — APPROVED** [review]

**Pull request merged**